drivers/lis2dh12/lis2dh12.c

```diff
@@ -68,24 +76,89 @@ static void _write(const lis2dh12_t *dev, uint8_t reg, uint8_t data)
     spi_transfer_reg(BUS, CS, reg, data);
 }
 
+/* and now the I2C specific part of the driver */
+#else
+
+/* I2C config */
+#define BUS_CLK             I2C_SPEED_FAST
+#define BUS_OK              (0)
+/* I2C shortcuts */
+#define BUS                 (dev->p->i2c)
+#define ADDR                (dev->p->addr)
+/* flag for enabling address auto-incrementation */
+#define FLAG_AINC           (0x80)
+
+static int _init_bus(const lis2dh12_t *dev)
+{
+    /* for I2C, we initialize the bus in master mode */
+    if (i2c_init_master(BUS, BUS_CLK) != BUS_OK) {
+        return LIS2DH12_NOBUS;
+    }
+    return LIS2DH12_OK;
+}
+
+static int _acquire(const lis2dh12_t *dev)
+{
+    return i2c_acquire(BUS);
+}
+
+static void _release(const lis2dh12_t *dev)
+{
+    i2c_release(BUS);
+}
+
+static uint8_t _read(const lis2dh12_t *dev, uint8_t reg)
+{
+    uint8_t tmp;
+    i2c_read_reg(BUS, ADDR, reg, &tmp);
+    return tmp;
+}
+
+static void _read_burst(const lis2dh12_t *dev, uint8_t reg,
+                              void *data, size_t len)
+{
+    i2c_read_regs(BUS, ADDR, (FLAG_AINC | reg), data, (int)len);
+}
+
+static void _write(const lis2dh12_t *dev, uint8_t reg, uint8_t data)
+{
+    DEBUG("[lis2dh12] write: reg 0x%02x, val 0x%02x\n", (int)reg, (int)data);
+    i2c_write_reg(BUS, ADDR, reg, data);
+}
+
+#endif  /* MODULE_LIS2DH12_SPI */
+
+
 int lis2dh12_init(lis2dh12_t *dev, const lis2dh12_params_t *params)
 {
     assert(dev && params);
 
     dev->p = params;
     dev->comp = (1000UL * (0x02 << (dev->p->scale >> 4)));
 
-    /* start by setting up the chip select pin */
-    if (spi_init_cs(BUS, CS) != SPI_OK) {
-        DEBUG("[lis2dh12] error: unable to initialize CS pin\n");
+#ifdef MODULE_LIS2DH12_SPI
+    /* initialize the chip select line */
+    if (_init_bus(dev) != LIS2DH12_OK) {
+        DEBUG("[lis2dh12] error: unable to initialize the CS pin\n");
         return LIS2DH12_NOBUS;
     }
+#endif
 
-    /* acquire the SPI bus and verify that our parameters are valid */
+    /* acquire the bus and verify that our parameters are valid */
     if (_acquire(dev) != BUS_OK) {
-        DEBUG("[lis2dh12] error: unable to acquire SPI bus\n");
+        DEBUG("[lis2dh12] error: unable to acquire the bus\n");
+        return LIS2DH12_NOBUS;
+    }
+
+#ifndef MODULE_LIS2DH12_SPI
+    /* in the current state, the I2C driver expects the bus to be acquired
+     * before the init function is called, so we have to switch order */
```

Comment:
So why not call `_acquire()` (and `_release()`) and inside `_init_bus()`  and save yourself the code duplication here.